### PR TITLE
Add random order option

### PR DESCRIPTION
**Pull request by chrissawyerfan4**
I recently wanted to just browse a little through my link collection and noticed there was no *order by random* option. Just added it, thought I'd contribute back the addition.

Let me know if I'm not using the language system correctly (maybe Spain now gets an error for missing key? I assume it will fall back) or if I should improve something else!

---

As an aside, I found the existing sort strings fairly hard to parse as a user. Here's a screenshot of how I changed them (I'm not married to that capitalization yet, I think it might be more readable but... it's also uglier so... hm). If you like, I could do another pull request with these strings for English and German. Or feel free to just copy it, in whole or in part, without doing the pull request dance first.  
creation date ascending became oldest, creation date descending became newest, etc., with the random option added

### CODE REVIEW

**Comment by chrissawyerfan4** [comment]
Thanks! It has been done :)

**Comment by welrbraga** [comment]
It is not working for me. I've just upgraded my instalation from 1.11.4 to 1.12.0 and everything is working nice except this option (for now).

When I selected "random" option, I have receveid a red message about "500 - Server Error" and and my log register a message about <Unknown column 'random'>


image


`error	production	2023-04-13 18:52:15	SQLSTATE[42S22]: Column not found: 1054 Unknown column 'random' in 'order clause' (SQL: select * from `links` where `is_private` = 0 and `links`.`deleted_at` is null order by `random` desc limit 24 offset 0) {"exception":"[object] (Illuminate\\Database\\QueryException(code: 42S22): SQLSTATE[42S22]: Column not found: 1054 Unknown column 'random' in 'order clause' (SQL: select * from `links` where `is_private` = 0 and `links`.`deleted_at` is null order by `random` desc limit 24 offset 0) at /app/vendor/laravel/framework/src/Illuminate/Database/Connection.php:760)`

**Comment by chrissawyerfan4** [comment]
@welrbraga could you post the full stack trace please? I'm eyeing my local git commit as well as the diff in this pull request, and it looks rather similar. It would help to know from which file this originates (like, what call stack ends up in this `Connection.php`) to narrow down where to search for the problem.

File: app/Http/Controllers/Models/LinkController.php [hunk 1]
```diff
@@ -31,11 +31,14 @@ public function index(Request $request): View
 
         $links = Link::byUser()
             ->with('tags')
-            ->orderBy($orderBy, $orderDir)
-            ->paginate(getPaginationLimit());
+            ->orderBy($orderBy, $orderDir);
```
Comment (Kovah) [inline app/Http/Controllers/Models/LinkController.php:34]: This should be put into an `else` to the statement below. When ordering by random, this would order by the `random` column which does not exists.

Comment (chrissawyerfan4) [reply app/Http/Controllers/Models/LinkController.php:34]: I was eyeing that line as well, though it works for me (in my local `master`, that's still how it works) so I was now digging into whether I made any other changes, but it doesn't look like it. Putting it in an `else` is probably the most sensible thing to do here. I'll open up a new pull request. (I can't simply append to this one because it's already merged, right?)

Comment (chrissawyerfan4) [reply app/Http/Controllers/Models/LinkController.php:34]: One difference is probably the database engine: I'm using sqlite. But that also isn't a fan of unknown columns in ordering clauses so it's not an explanation by itself.

```
sqlite> select * from users order by blaat;
Error: no such column: blaat
sqlite> 
```
